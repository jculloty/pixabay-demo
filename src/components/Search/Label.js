import React from 'react';
import PropTypes from 'prop-types';

import { camalCaseToDisplayedName } from 'services/utils';

const Label = function(props) {
  const { name } = props;
  return <label htmlFor={name}>{camalCaseToDisplayedName(name)}</label>;
};

Label.propTypes = {
  name: PropTypes.string.isRequired,
};

export default Label;
