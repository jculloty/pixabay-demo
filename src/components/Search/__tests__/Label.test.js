import React from 'react';
import renderer from 'react-test-renderer';

import Label from '../Label';

const buildComponent = function(name = 'testingOneTwoThree') {
  return (<Label name={name} />);
};

describe('Search: Label', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(buildComponent())
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
