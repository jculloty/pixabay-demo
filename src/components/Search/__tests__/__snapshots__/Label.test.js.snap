// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search: Label renders correctly 1`] = `
<label
  htmlFor="testingOneTwoThree"
>
  Testing One Two Three
</label>
`;
